concepts/class-inheritance/about.md

```diff
@@ -1,2 +1,291 @@
+           print("Name: ", self.__name)
+           print("Roll: ", self.__roll)
+           print("Branch: ", self.__division)
+    
+     # public member function
+     def accessPrivateFunction(self):
+            
+           # accessing private member function
+           self.__displayDetails() 
+ 
+# creating object   
+obj = Student("ABC", 122333, "Physics")
+ 
+# accessing public member function of the class
+obj.accessPrivateFunction()
+```
+`__name`, `__roll`, and `__division` are private members of the class `Student`, while the `__displayDetails()` method is a private member function (accessible only within the class), and the `accessPrivateFunction()` method is a public member function of the class `Student` that can be accessed from anywhere within the programme. The `accessPrivateFunction()` method gives us access to the `Student` class's private members.
+### Class Functions
+The member functions of a class can be categorized into following two categories:
+
+__1. Accessor Functions__: These are those number functions that allow us to access data fields of the object. However, these functions cannot/do not change the value of the data members.
+
+```python
+class Student:
+    # private members
+     __name = None
+     __roll = None
+     __marks = None
+     __div = None
+     
+     # constructor
+    def __init__(self, name, roll, marks, div): 
+        self.__name = name
+        self.__roll = roll
+        self.__marks = marks
+        self.__div = div
+     
+     # public member function 
+     def displayDetails(self):
+         # accessing private data members
+         print("Name: ", self.__name)
+         print("Roll: ", self.__roll)
+         print("Marks:", self.__marks)
+         print("Division: ", self.__div)
+         
+     # Accessor Function    
+     def getRollNo(self):
+        return self.__roll
+     
+     # Mutator Function
+     def setDiv(self):
+        if self.__marks>=60:
+            self.__div = 'A'
+        elif self.__marks<60:
+            self.__div = 'B'
+        return self.__div
+```
+__2. Mutator Function__: This method is used to change the state of an object, i.e. it changes the data variable's hidden value. It has the ability to change the value of a variable on the fly. The update method is another name for this function. We can also use the word _set_ to name these procedures.
+
+As per the above example, `getRollNo()` is an accessor function and `setDiv` is a mutator function.
+
+## Inheritance
+
+## Single Inheritance
+
+When a subclass inherits only from one base class, it is known as single inheritance.
+
+```python
+#Base class
+class Odin:
+    def func1(self):
+        print("Odin: father of thor")
+
+#Derived/Subclass class
+class Thor(Odin):
+    def func2(self):
+        print("Thor: son of Odin")
+
+
+object = Thor()
+object.func1()
+object.func2()
+```
+The derived class will print both the statements of `func1` and `func2`.

```

Comment:
I think this needs to be a more direct and functional example that uses both attributes and methods, to demonstrate _what_ gets inherited, and _how_ one then adds to or overrides those properties and functionality in the child class.  Recommend taking a look at [this DO article](https://www.digitalocean.com/community/tutorials/understanding-class-inheritance-in-python-3) for some inspiration and examples.   This [real python](https://realpython.com/python3-object-oriented-programming/#inherit-from-other-classes-in-python) article is good as well.

Reply:
You are right, I was also getting a feeling that the example wasn't upto the mark. I have written a new example.